std-bits/table/src/main/java/org/enso/table/util/NameDeduplicator.java

```diff
@@ -1,29 +1,73 @@
 package org.enso.table.util;
 
+import org.enso.table.problems.Problem;
+import org.enso.table.util.problems.DuplicateNames;
+import org.enso.table.util.problems.InvalidNames;
+
+import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class NameDeduplicator {
-  public static List<String> deduplicate(List<String> names) {
-    return deduplicate(names, " ");
+  private final Set<String> usedNames = new HashSet<>();
+  private final List<String> invalidNames = new ArrayList<>();
+  private final List<String> duplicatedNames = new ArrayList<>();
+
+  public String makeValid(String input) {
+    if (input == null || input.isEmpty()) {
+      this.invalidNames.add(input);
+      return "Column";
```

Comment:
Are you sure we want to bake this into the deduplicator? What if we use it for something else in the future?
I'd suggest making this a constructor parameter (possibly with a default value).